src/k8s/pkg/k8sd/api/endpoints.go

```diff
@@ -155,6 +155,11 @@ func (e *Endpoints) Endpoints() []rest.Endpoint {
 			Path: apiv1.ClusterAPICertificatesRunRPC,
 			Post: rest.EndpointAction{Handler: e.postRefreshCertsRun, AccessHandler: e.ValidateNodeTokenAccessHandler("node-token"), AllowUntrusted: true},
 		},
+		{
+			Name: "ClusterAPI/RefreshCerts/Approve",
+			Path: apiv1.ClusterAPIApproveWorkerCSRRPC,
+			Post: rest.EndpointAction{Handler: e.postApproveWorkerCSR, AccessHandler: ValidateCAPIAuthTokenAccessHandler("capi-auth-token"), AllowUntrusted: true},
```

Comment:
I guess we should have this `capi-auth-token` in the `k8s-snap-api` as well. Created a card for this.